Fix IOS sendMail argument and allocation/crash

CircuitCipher opened the pull request:
The ios sendMail function was using the first Lua argument for
recipient, message and subject.  Updated this to use the correct
arguments.

Also in sendMail, a crash was occuring for me.  This seemed to be due
to an array being alloc'ed but constructed with an autorelease method.
I removed the alloc and just created the array using the autorelease
method and this fixed the crash.

Commit:
Fix IOS sendMail argument and allocation/crash

The ios sendMail function was using the first Lua argument for
recipient, message and subject.  Updated this to use the correct
arguments.

Also in sendMail, a crash was occuring for me.  This seemed to be due
to an array being alloc'ed but constructed with an autorelease method.
I removed the alloc and just created the array using the autorelease
method and this fixed the crash.

jeiting commented on src/moaiext-iphone/MOAIAppIOS.mm:
```diff
@@ -147,13 +147,13 @@
 	controller.mailComposeDelegate = MOAIAppIOS::Get ().mMailDelegate;
 	
-	NSArray* to = [[ NSArray alloc ] arrayByAddingObject:[[ NSString alloc ] initWithUTF8String:recipient ]];
+	NSArray* to = [ NSArray arrayWithObject:[[ NSString alloc ] initWithUTF8String:recipient ]];
```
All of these [[NSString alloc] initWithUTF8String] are leaking. They should be wrapped with an autorelease.

CircuitCipher commented:
Thanks for the review Jacob. I'll address those in the next commit.

Commit:
Fixiing additional memory leaks in sendMail

Fixing the additional memory leaks pointed out by Jacob in the Pull
Request by changing the allocation of the NSString to be autoreleased.
